vm/devices/user_driver/src/emulated.rs

```diff
@@ -76,6 +78,9 @@ impl<T: PciConfigSpace + MmioIntercept> EmulatedDevice<T> {
         let controller = MsiController::new(msi_set.len());
         msi_set.connect(&controller);
 
+        let bars = device.probe_bar_masks();
+        let bar0_len = !(bars[0] & !0xf) as usize + 1;
+
         // Enable BAR0 at 0, BAR4 at X.
         device.pci_cfg_write(0x20, 0).unwrap();
```

Comment:
Just curious, since I'm late to the party. Do we have consensus in openvmm if we eventually want to move these hex numbers to consts? e.g. use `pci_core::cfg_space::HeaderType00::BAR4`?

Just asking the question, in a way I prefer the hex magic values because it's easier to follow along when reading a spec.

Reply:
I think we'll probably want to move to the constants; we generally do so in product code, but not always in test code.

But we'll also want to extend this test code to support other BARs and to query config space to determine where its MSI-X tables live rather than assume they're on BAR 4. I figured I'd leave that kind of cleanup for whoever wants to generalize this test code.